### ISSUE TITLE
Errors updating subscription items quantity in Stripe

### ISSUE THREAD

**Issue by kimili** [pay-rails/pay#725]
I appreciate the new(ish) subscription items support for Stripe, but I've found a few holes in the implementation, specifically around the `change_quantity` implementation:

## `change_quantity` on `Pay::Subscription` only accepts one argument.

When I have a subscription that has multiple subscription items set up on an account with Stripe as the payment processor, this error is thrown when I try to change the quantity on one of the items:

```
account.subscription.change_quantity(12, subscription_item_id: item_id)
  Pay::Subscription Load (0.6ms)  SELECT "pay_subscriptions".* FROM "pay_subscriptions" WHERE "pay_subscriptions"."customer_id" = $1 AND "pay_subscriptions"."name" = $2 ORDER BY "pay_subscriptions"."id" DESC LIMIT $3  [["customer_id", 17], ["name", "default"], ["LIMIT", 1]]
/Users/michael/.asdf/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/pay-5.0.4/app/models/pay/subscription.rb:140:in `change_quantity': wrong number of arguments (given 2, expected 1) (ArgumentError)
```

As I understand it, that method should ideally pass additional parameters over to the the payment processor's `change_quantity` method. I know only Stripe has that at the moment, but it would be nice to be able to have that not fail. In the meantime, I've worked around this particular issue like so:

```
stripe_subscription = account.subscription.class.pay_processor_for('stripe').new(account.subscription)
stripe_subscription.change_quantity(quantity, subscription_item_id: item_id)
```

where it would be much nicer to just do

```
account.subscription.change_quantity(12, subscription_item_id: item_id)
```

---

## `change_quantity` on `Pay::Stripe::Subscription` throws an API error

That method calls [`::Stripe::SubscriptionItems.update`](https://stripe.com/docs/api/subscription_items/update?lang=ruby) with the id of the subscription item to update, and a hash that includes the quantity _and the id of the subscription item to update_. Something like this:

`Stripe::SubscriptionItems.update(item_id, { quantity: 11, subscription_item_id: item_id })`

Doing that throws an API error like so:

```
/Users/michael/.asdf/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/stripe-7.1.0/lib/stripe/stripe_client.rb:714:in `handle_error_response': Received unknown parameter: subscription_item_id (Stripe::InvalidRequestError)
```

When I remove the `subscription_item_id` from the options hash, so the call look something like `Stripe::SubscriptionItems.update(item_id, { quantity: 11 })`, it works correctly:

I suppose the crude fix for this would be to do an `options.delete(:subscription_item_id)` in the `change_quantity` method, like this, but there's probably a more elegant way to fix this.

```
      def change_quantity(quantity, **options)
        subscription_item_id = options.fetch(:subscription_item_id, subscription_items&.first&.dig("id"))
        if subscription_item_id
          options.delete(:subscription_item_id)
          ::Stripe::SubscriptionItem.update(subscription_item_id, options.merge(quantity: quantity), stripe_options)
          @stripe_subscription = nil
        else
          @stripe_subscription = ::Stripe::Subscription.update(processor_id, options.merge(quantity: quantity).merge(expand_options), stripe_options)
        end
        true
      rescue ::Stripe::StripeError => e
        raise Pay::Stripe::Error, e
      end
```

Also while I haven't tested them, in the same `Pay::Stripe::Subscription` class, I noticed that the `create_usage_record` and `usage_record_summaries` methods include the `subscription_item_id` in the `options` hash that they send to the Stripe API. I'm not sure, but I wouldn't be surprised if those calls similarly fail for the same reason.

**Comment by excid3** [comment]
That's a bug. `change_quantity` should also forward options to the payment processors.

**Comment by kimili** [comment]
Thanks for this @excid3, but the patch doesn't include addressing the API error that Stripe throws when you pass `subscription_item_id` as an option to `::Stripe::SubscriptionItem.update`

Basically, all it needs is this:

```
::Stripe::SubscriptionItem.update(subscription_item_id, quantity: quantity)
```

**Comment by excid3** [comment]
That should be fixed in 94bd889 now
